test/random/README.md

```diff
@@ -0,0 +1,157 @@
+| testSetOffchainRandomSourceBadAuth | setOffchainRandomSource() without authorization. | No       | Yes         |
+| testSetRanDaoAvailable          | setRanDaoAvailable().                             | Yes        | Yes         |
+| testSetRanDaoAvailableBadAuth   | setRanDaoAvailable() without authorization.       | No         | Yes         |
+| testAddOffchainRandomConsumer   | addOffchainRandomConsumer().                      | Yes        | Yes         |
+| testAddOffchainRandomConsumerBadAuth | addOffchainRandomConsumer() without authorization.| No    | Yes         |
+| testRemoveOffchainRandomConsumer| removeOffchainRandomConsumer().                   | Yes        | Yes         |
+| testRemoveOffchainRandomConsumerBadAuth | removeOffchainRandomConsumer() without authorization.| No | Yes      |
+| testUpgrade                     | Check that the contract can be upgraded.          | Yes        | Yes         |
+| testUpgradeBadAuth              | Check upgrade authorisation.                      | No         | Yes         |
+| testNoUpgrade                   | Upgrade from V0 to V0.                            | No         | Yes         |
+
+
+Operational functions tests:
+
+| Test name                       |Description                                        | Happy Case | Implemented |
+|---------------------------------| --------------------------------------------------|------------|-------------|
+| testTradNextBlock               | Check basic request flow                          | Yes        | Yes         |
+| testRanDaoNextBlock             | Check basic request flow                          | Yes        | Yes         |
+| testOffchainNextBlock           | Check basic request flow                          | Yes        | Yes         |
+| testOffchainNotReady            | Attempt to fetch offchain random when not ready   | No         | Yes         |
+| testTradTwoInOneBlock           | Two calls to requestRandomSeed in one block       | Yes        | Yes         |
+| testRanDaoTwoInOneBlock         | Two calls to requestRandomSeed in one block       | Yes        | Yes         |
+| testOffchainTwoInOneBlock       | Two calls to requestRandomSeed in one block       | Yes        | Yes         |
+| testTradDelayedFulfilment       | Request then wait several blocks before fulfilment | Yes       | Yes         |
+| testRanDaoDelayedFulfilment     | Request then wait several blocks before fulfilment | Yes       | Yes         |
+
+Scenario: Generate some random numbers, switch random generation methodology, generate some more
+numbers, check that the numbers generated earlier are still available:
+
+| Test name                       |Description                                        | Happy Case | Implemented |
+|---------------------------------| --------------------------------------------------|------------|-------------|
+| testSwitchTraditionalOffchain   | Traditional -> Off-chain.                         | Yes        | Yes         |
```

Comment:
nit: suggest replacing "traditional" here (and other places like test function names etc.) to "onchain" given the former does not appear to be used in the main contracts any longer.

Reply:
For the situations in which it is just generic "onchain", i have switched the language.